elyra/pipeline/catalog_connector.py

```diff
@@ -90,7 +143,7 @@ def get_catalog_entries(self, catalog_metadata: Dict[str, Any]) -> List[Dict[str
                     }
 
         :returns: a list of catalog entry dictionaries, each of which contains the information
-                  needed to access a component definition in read_catalog_entry()
+                  needed to access a component definition in get_component_definition()
         """
         raise NotImplementedError(
             "abstract method 'get_catalog_entries()' must be implemented"
```

Comment:
If we leave the `@abstractmethod` decorator in the code then all implementing classes must provide an implemention. To avoid that we should remove the decorator.

Reply:
Was this comment intended for `read_catalog_entry()` below?  We still want `get_catalog_entries()` to be an abstract method.

FWIW, the behavior you describe only occurs when the class derives from the ABC meta-class.  Since that "thunking" is not happening here, these aren't true ABCs - where enforcement requires implementation.  Instead, the enforcement happens at the time of the call.  (True abstract methods don't require a `NotImplementedError`, they can even be a `pass` statement.) Kiersten and I talked about that and there was some concern that some unwitting future maintainer will see that and convert these to true abstract classes at which time, we would require implementations - so taking the approach of `@deprecated` is definitely preferred.  Unfortunately, abstract methods for publicly consumable classes only work for the initial implementation.

(Our [SchemasProvider class](https://github.com/elyra-ai/elyra/blob/647351101ed36c2c2ad9210c3bfb66a1112cbd90/elyra/metadata/schema.py#L379-L390) is an example that sets up the ABC in a manner that enforces implementation at instantiation.)